File: src/Helpers/folder.py
class FolderSubItem:
    """
    Represents a basic folder with attributes like name, path, identifier, and parent_id.
    """

    def __init__(self, name, path, identifier, parent_id=None):
        self.name = name
        self.path = path
        self.identifier = identifier
        self.parent_id = parent_id

    def __repr__(self):
        return f"FolderSubItem(name={self.name}, path={self.path})"


class Folder:
    """
    Represents a folder with separate source and destination attributes.
    """

    def __init__(self, source: FolderSubItem = None, destination: FolderSubItem = None):
        self.source = source
        self.destination = destination

    def __repr__(self):
        return (
            f"Folder(Source={self.source}, "
            f"Destination={self.destination})"
        )

File: code/core/filesystem/folder.go
package filesystem

import "fmt"

// Folder represents a basic folder with attributes like name, path, identifier, and parentID.
type Folder struct {
	Name          string
	Path          string
	Identifier    string
	ParentID      string 
	LastModified  string 
	Level         int
}

// String method to provide a formatted representation of Folder (like __repr__)
func (f Folder) String() string {
	return fmt.Sprintf("Folder(name=%s, path=%s, lastModified=%s, level=%v)", f.Name, f.Path, f.LastModified, f.Level)
}
